### PR TITLE
refactor(components): [message] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by yuanxin518**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/message/src/message.vue [hunk 1]
```diff
@@ -55,94 +55,72 @@ import type { BadgeProps } from '@element-plus/components/badge'
 
 import type { CSSProperties } from 'vue'
 
-export default defineComponent({
+// export default defineComponent({
```
Comment (JeremyWuuuuu) [inline packages/components/message/src/message.vue:58]: **Suggestion:** delete this line.

File: packages/components/message/src/message.vue [hunk 2]
```diff
@@ -55,94 +55,72 @@ import type { BadgeProps } from '@element-plus/components/badge'
...
+  }
+)
 
-      close,
-      clearTimer,
-      startTimer,
-    }
-  },
-})
+useEventListener(document, 'keydown', keydown)
```
Comment (JeremyWuuuuu) [inline packages/components/message/src/message.vue:125]: **Suggestion:** put line#125 inside line#112 `onMounted` body. Although this is not what you made,
but just for correctness. 

Remember to ask me about refactoring a component like https://github.com/element-plus/element-plus/issues/6114#issuecomment-1125552539

Next time, PRs like these will not be accepted if you did not ask in advance.

**Review by jw-foss — COMMENTED** [review]
@Huangyuanx Thank you for contributing, left a few comments, back to you 🏓

**Comment by Huangyuanx** [comment]
@JeremyWuuuuu. I've modified them.

File: packages/components/message/src/message.vue [hunk 3]
```diff
@@ -55,94 +55,70 @@ import type { BadgeProps } from '@element-plus/components/badge'
...
   name: 'ElMessage',
+})
+const props = defineProps(messageProps)
+const emits = defineEmits(messageEmits)
+const ns = useNamespace('message')
+const visible = ref(false)
+const badgeType = ref<BadgeProps['type']>(
+  props.type ? (props.type === 'error' ? 'danger' : props.type) : 'info'
+)
+let stopTimer: (() => void) | undefined = undefined
```
[the remark this answers is not in the document]
Comment (Huangyuanx) [reply packages/components/message/src/message.vue]: ok. I modified it.

**Pull request closed without merging**